pkg/prom/ha/http_test.go

```diff
@@ -156,6 +162,50 @@ func TestServer_PutConfiguration(t *testing.T) {
 	}
 }
 
+func TestServer_PutConfiguration_NonUnique(t *testing.T) {
+	env := newAPITestEnvironment(t)
+
+	conflictA := instance.DefaultConfig
+	conflictA.Name = "conflict-a"
+	conflictA.HostFilter = false
+	conflictA.RemoteFlushDeadline = 10 * time.Minute
+	conflictA.ScrapeConfigs = []*config.ScrapeConfig{{
+		JobName:     "conflicting",
+		MetricsPath: "/metrics",
+		Scheme:      "http",
+	}}
+	_ = conflictA.ApplyDefaults(&config.DefaultGlobalConfig)
+
+	//
+	// Put conflict A; it should succeed
+	//
+	bb, err := yaml.Marshal(conflictA)
+	require.NoError(t, err)
+
+	resp, err := http.Post(env.srv.URL+"/agent/api/v1/config/conflict-a", "", bytes.NewReader(bb))
+	require.NoError(t, err)
+	require.Equal(t, http.StatusCreated, resp.StatusCode)
+	unmarshalTestResponse(t, resp.Body, nil)
+
+	//
+	// Put conflict B; it should fail because conflict-a already
+	// has a job with the name test_job
```

Comment:
this should be "conflicting" instead of "test_job", no?

Reply:
+1

Reply:
Yes, thank you!